### PR TITLE
core(tsc): add type checking to sentry usage

### LINKED ISSUE #5971: Sample our sentry errors more heavily
we have a lot of high volume errors that are just happening too often.

Here's our top 25 errors in the last 24hr:

```
"Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules": 305k,
"Could not find any top level events": 249k,
"No firstMeaningfulPaint found, usingfallback: 214k,
"Invalid task timing data": 202k,
"The asynchronous expression exceeded the allotted time of 60s": 18k,
"Could not load stylesheet (timeout)": 104k,
"Invalid Version: <version>": 103k,
"Invalid image sizing information <url>": 76k,
"Cannot create DNS cache with no rtt": 70k,
"finalUrl <url> not found in network records.": 50k,
"SPEEDINDEX_OF_ZERO": 34k,
"NO_TTI_NETWORK_IDLE_PERIOD": 18k,
"NO_FMP": 33k,
"Images have different sizes!": 29k,
"Cannot simulate graph with cycle": 28k,
"Cannot read property 'indexOf' of undefined": 24k,
"Cannot read property 'after' of undefined": 16k,
"Failed to construct 'URL': Invalid URL": 14k,
"navigationStart was not found in the trace": 14k,
"Failed to execute 'matches' on 'Element': '<selector>' is not a valid selector.": 13k,
"Failed to execute 'getComputedStyle' on 'Window': parameter 1 is not of type 'Element'.": 11k,
"NO_FCPUI_IDLE_PERIOD": "106k,
"NO_FCP": "102k
```

in sentry.js you'll see we sample some of these. we could afford to sample more... well actually, it probably makes sense to sample everything. lets start at 1%?

### PR DESCRIPTION
**Pull request by brendankenny**
long-standing TODO.

Also fixes #5971

### CODE REVIEW

File: lighthouse-core/lib/sentry.js [hunk 1]
```diff
@@ -3,94 +3,106 @@
...
-  {pattern: /^NO_.*/, rate: 0.1},
   // Message based sampling
-  {pattern: /Could not load stylesheet/, rate: 0.01},
-  {pattern: /Failed to decode/, rate: 0.1},
-  {pattern: /All image optimizations failed/, rate: 0.1},
-  {pattern: /No.*resource with given/, rate: 0.01},
-  {pattern: /No.*node with given id/, rate: 0.01},
+  {pattern: /Could not load stylesheet/, rate: 0.1},
+  {pattern: /No.*resource with given/, rate: 0.1},
+  {pattern: /No.*node with given id/, rate: 0.1},
```
Comment (brendankenny) [inline lighthouse-core/lib/sentry.js]: we could drop these completely? Also do they fire multiple times per run? We could fire once per audit instead (does sentry support adding a count payload and have a decent aggregation view of them?)

File: lighthouse-core/lib/sentry.js [hunk 2]
```diff
@@ -3,94 +3,95 @@
...
+    // Special case captureException to return a Promise so we don't process.exit too early
+    sentryDelegate.captureException = async (err, opts = {}) => {
       // Ignore if there wasn't an error
-      if (!err) return empty;
+      if (!err) return;
+
       // Ignore expected errors
-      if (err.expected) return empty;
-      // Sample known errors that occur at a high frequency
-      const sampledErrorMatch = SAMPLED_ERRORS.find(sample => sample.pattern.test(err.message));
```
Comment (paulirish) [inline lighthouse-core/lib/sentry.js]: can we keep this block?

along with something like this above:

```js
const SAMPLED_ERRORS = [	
  // Error code based sampling	
  // e.g.: {pattern: /No.*node with given id/, rate: 0.01},	
];
```

**Comment by paulirish** [comment]
type stuff and fonts look good!

File: lighthouse-core/lib/sentry.js (hunk 2 above)
Comment (brendankenny) [reply lighthouse-core/lib/sentry.js]: There you go :P

File: lighthouse-core/lib/sentry.js [hunk 3]
```diff
@@ -3,94 +3,105 @@
...
+
 /**
- * We'll create a delegate for sentry so that environments without error reporting enabled will use
+ * A delegate for sentry so that environments without error reporting enabled will use
  * noop functions and environments with error reporting will call the actual Sentry methods.
  */
-const sentryDelegate = Object.assign({}, sentryApi);
-sentryDelegate.init = function init(opts) {
+const sentryDelegate = {
+  init,
```
Comment (paulirish) [inline lighthouse-core/lib/sentry.js]: lol the hoist. a lil fancy for my tastes, but ok.. :)

**Pull request merged**